spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplateIntegrationTests.java

```diff
@@ -0,0 +1,57 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.core;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.cloud.gcp.data.spanner.AbstractSpannerIntegrationTest;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerTable;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * @author Balint Pato
+ */
+
+@RunWith(SpringRunner.class)
+public class SpannerTemplateIntegrationTests extends AbstractSpannerIntegrationTest {
+
+	@Test
```

Comment:
cool set up ! 
are the setupAndClean() and clean() methods called before and after each test? These tests don't necessarily run in sequence according to junit right?

Reply:
yes, @After runs after each test and @Before before each test. Junit tests orders are not guaranteed.

Reply:
should we then use unique names for tables for each test since one might forget to clear the table after writing each test or if two tests run at the same time?

Reply:
yeah, I'll make that change - it's a bit messy though, as we'll need to update the table names in `@SpannerTable` annotations.